src/sinks/datadog/metrics/encoder.rs

```diff
@@ -510,10 +564,89 @@ fn encode_timestamp(timestamp: Option<DateTime<Utc>>) -> i64 {
     }
 }
 
+// Given the vector source type, return the OriginService value associated with that integration.
+//
+// Some sources such as `kafka`, `nats`, `redis` for example, are only capable of receiving metrics
+// with the `native` or `native_json` codec. In such cases we intentionally do not set the origin
+// metadata here, because the true origin will have already been determined to be a pass-through.
+fn source_type_to_service(source_type: &'static str) -> Option<u32> {
```

Comment:
We did actually talk about this but the thinking was that we wanted to try to isolate the impact of this change to the `datadog_metrics` sink since that is the one that really needs it rather than leak it into source components. I do see what you are saying about the risk of forgetting to add it to a new metrics source. I wonder if we could also resolve that through testing?

Reply:
I don't think a warning log would be an appropriate signal for this because this isn't really a user error or runtime error.